STYLE-GUIDE.md

```diff
@@ -17,3 +17,179 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
 * Table names should be pluralized.  This is to avoid *most* used words. e.g. orders instead of order
 * Table names with multiple words should be snake_case.
 * Column names with multiple words should be merged together.  e.g. `getFirstName()` would map to `firstname`
+
+# Unit tests 
+* __Every single test should be independent and isolated. Unit test shouldn't deped on another unit test.__ 
+
+  There is one incorrect example below:
+  ```java
+  @Test
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    age = 17;
+    boolean isAdult = ageService.isAdult(age);
+    assertEquals(false, isAdult);
+    testIsNotAnAdultIfAgeMoreThan18();
```

Comment:
What I understand with dependant tests is that the result of one depends on the execution of the other one. Not that they directly invoke each other. Eg. we keep a list of items and we add an item in the first test, asserting that the size is 1, then we add a new item in another test and assert that the size is 2 (which depends on whether the first one executed)